odoo/models.py

```diff
@@ -2947,15 +2947,20 @@ def _field_to_sql(self, alias: str, fname: str, query: (Query | None) = None, fl
             return SQL("COALESCE(%s)", SQL(", ").join(sql_field_langs))
 
         if field.company_dependent:
-            fallback = field.get_company_dependent_fallback(self)
-            fallback = field.convert_to_column(field.convert_to_write(fallback, self), self)
             sql_field = SQL(
-                "COALESCE(%(column)s->%(company_id)s,to_jsonb(%(fallback)s::%(column_type)s))",
+                "%(column)s->%(company_id)s",
                 column=sql_field,
                 company_id=str(self.env.company.id),
-                fallback=fallback,
-                column_type=SQL(field._column_type[1]),
             )
+            fallback = field.get_company_dependent_fallback(self)
+            fallback = field.convert_to_column(field.convert_to_write(fallback, self), self)
+            if fallback not in (None, 0):  # 0, 0.0, False, None
```

Comment:
I'm not sure but should string be added as well?
```suggestion
            if fallback not in (None, 0, ''):  # 0, 0.0, False, None, ''
```

Reply:
Currently for `_String.convert_to_record`
`None` -> `False`
`''` -> `''`
NULL and '' are different logic values in DB